### DISCUSSION TITLE
Improving 3D segmentation mean dice

### DISCUSSION THREAD

**Discussion by eanemo** [Project-MONAI/MONAI#2905] [Q&A]
Hi,

I am trying to train a model for a binary 3D segmentation task. I have created a custom dataset that includes 46 volumes (nii.gz format) for training and 9 for testing. I have used the [example](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/unet_segmentation_3d_ignite.ipynb) provided in the tutorial section as the base code for training. 
I have used several networks architectures (UNet, VNet, SegResNet, etc), solvers (Adam, AdamW, etc) and values for learning rate param (0.001, 0.0001, etc). The best model that I have obtained so far has a mean dice value of 0.65 and loss is about 0.12. 
I have tested the source code by using a testing dataset created using the create_test_image_3d method from Monai. Using this testing dataset I get a segmentation model with a mean dice value over 0.95 without any problems. So I presume that the source code is correct.
My custom dataset contains images like the following ones:

Image
![image](https://user-images.githubusercontent.com/70317462/132304386-e866b227-bec3-4333-9fbb-b857d0808ac4.png)

Segmentation (Ground Truth)
![image](https://user-images.githubusercontent.com/70317462/132304522-293c6047-445c-4e41-bbf2-202470d8dda8.png)

The model obtain a segmentation like this one:
![image](https://user-images.githubusercontent.com/70317462/132304768-90c45552-ce1d-43b1-9b46-556bfbf47ab7.png)

Any idea, suggestion for improving my current results?

**Comment by Nic-Ma** [comment]
Hi @dongyang0122 @wyli ,

Could you please help share some advice about this ticket?
I feel the image & label resolution is too low and cause the model hard to predict.

Thanks.

**Reply by eanemo** [reply]
The image resolution used in this training process is 256x256x256. In the examples directory, I have seen some [code](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/ignite/unet_training_array.py) that uses smaller image resolution (128x128x128) without any problem. So, I think that resolution of the image in the dataset should not be the problem.

**Reply by dongyang0122** [reply]
The "resolution" in the context of medical imaging denotes spacing (physical size of pixels/voxels). You can want to check the transform SpacingD to further increase it.

**Reply by eanemo** [reply]
Ok. I will try to increase the resolution of the image to check if this is the problem.

**Comment by dongyang0122** [comment]
Hi @eanemo, can you share the usage of create_test_image_3d or parts of implementation?

**Reply by eanemo** [reply]
Hi, the code that generates the testing dataset is as following:

```
for i in range(20):
    img_filename = "image_%d_cut.nii.gz" % (i)
    label_filename = "label_%d_segmentation.nii.gz" % (i)
    noisy_im, label = create_test_image_3d(height=128, width=128, depth=128, noise_max=0.1, num_seg_classes=1)
    nii_im = nib.Nifti1Image(noisy_im, np.eye(4))
    nii_label = nib.Nifti1Image(label, np.eye(4))
    nib.save(nii_im, join(target_dir, img_filename))
    nib.save(nii_label, join(target_dir, label_filename))
```

**Reply by dongyang0122** [reply]
What do generated image and label look like?

**Reply by eanemo** [reply]
Images generated with the code above looks like the following ones:

Image
![image_0_cut](https://user-images.githubusercontent.com/70317462/132841943-3c2b5070-c2f4-47d6-b878-844aa304e294.png)

Segmentation
![label_0_segmentation](https://user-images.githubusercontent.com/70317462/132841968-03f30b44-5de5-4ded-acd6-08110001552b.png)
